src/main/java/com/faforever/client/fx/contextmenu/AddPlayerNoteMenuItem.java

```diff
@@ -0,0 +1,45 @@
+package com.faforever.client.fx.contextmenu;
+
+import com.faforever.client.domain.PlayerBean;
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.player.PlayerNoteController;
+import com.faforever.client.player.PlayerService;
+import com.faforever.client.theme.UiService;
+import com.faforever.client.ui.dialog.Dialog;
+import com.faforever.client.util.Assert;
+import javafx.scene.layout.StackPane;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+public class AddPlayerNoteMenuItem extends AbstractMenuItem<PlayerBean> {
+
+  private final UiService uiService;
+  private final PlayerService playerService;
+  private final I18n i18n;
+
+  @Override
+  protected void onClicked() {
+    Assert.checkNullIllegalState(object, "No player has been set");
+    PlayerNoteController playerNoteController = uiService.loadFxml("theme/player_note.fxml");
+    Dialog dialog = uiService.showInDialog((StackPane) getParentPopup().getOwnerWindow().getScene().getRoot(), playerNoteController.getRoot());
+    playerNoteController.setPlayer(object);
+    playerNoteController.setCloseButtonAction(event -> dialog.close());
+    dialog.setOnDialogOpened(event -> playerNoteController.requestFocus());
+    dialog.show();
+  }
+
+  @Override
+  protected boolean isItemVisible() {
+    return object != null && !playerService.containsNote(object);
+  }
+
+  @Override
+  protected String getItemText() {
+    return i18n.get("chat.userContext.addNote");
```

Comment:
Why are this and EditPlayerNote two separate objects? Why not just have a single class that gives different text depending on whether there is a player note that already exists for this player, as otherwise they are identical.